Gems/Atom/Tools/AtomToolsFramework/Code/Source/Document/AtomToolsDocumentMainWindow.cpp

```diff
@@ -114,7 +122,7 @@ namespace AtomToolsFramework
         m_menuFile->insertAction(insertPostion, m_actionSaveAsCopy);
 
         m_actionSaveAsChild = CreateAction("Save As &Child...", [this]() {
-            const AZ::Uuid documentId = GetDocumentTabId(m_tabWidget->currentIndex());
+            const AZ::Uuid documentId = GetCurrentDocumentId();
             const QString documentPath = GetDocumentPath(documentId);
 
             bool result = false;
```

Comment:
I don't think this will go away. I think the functions will probably be renamed something more generic and the text in the menu options can be configurable so that material editor can maintain its UX. All of the options can be renamed or hidden programmatically based on context. Other types of documents could keep track of a breadcrumb or save a completely different derived type.